src/network.zig

```diff
@@ -1277,6 +1277,95 @@ pub const Protocols = struct {
 			conn.send(.fast, id, writer.data.items);
 		}
 	};
+	pub const explode = struct {
+		pub const id: u8 = 14;
+		pub const asynchronous = false;
+		fn receive(conn: *Connection, reader: *utils.BinaryReader) !void {
+			const pos = try reader.readVec(Vec3i);
+			const strength = try reader.readFloat(f32);
+			const radius = try reader.readFloat(f32);
+
+			if(conn.user != null) {
+				if(!main.server.world.?.allowExplosives) return;
+				doExplode(pos, strength, radius);
+			} else {
+				return error.InvalidPacket;
+			}
+		}
+		fn doExplode(pos: Vec3i, _strength: f32, _radius: f32) void {
+			const radius = @min(@max(_radius, 0), 256);
+			const strength = @min(@max(_strength, 0), 1024);
+
+			const diameter: u32 = @intFromFloat(std.math.ceil(radius)*2);
+			const goldenAngle = std.math.pi*(3.0 - @sqrt(5.0));
+			const radiusInt: usize = @intFromFloat(std.math.ceil(radius));
+			const numSamples: usize = @intFromFloat(std.math.ceil(4*std.math.pi*radius*radius)*2);
+
+			const exploded = main.utils.Array3D(bool).init(main.stackAllocator, diameter, diameter, diameter);
+			defer exploded.deinit(main.stackAllocator);
+			for(exploded.mem) |*e| {
+				e.* = false;
+			}
+
+			for(0..numSamples) |i| {
+				const z = 1 - 2*@as(f32, @floatFromInt(i))/@as(f32, @floatFromInt(numSamples));
+				const radiusAtZ = @sqrt(1 - z*z);
+				const theta = goldenAngle*@as(f32, @floatFromInt(i));
+				const x = radiusAtZ*@cos(theta);
+				const y = radiusAtZ*@sin(theta);
+
+				var damage = strength;
+
+				for(0..radiusInt) |j| {
+					const distanceFactor = @as(f32, @floatFromInt(j))/@as(f32, @floatFromInt(radiusInt));
+					const distance = radius*distanceFactor;
+
+					const xOffset: i32 = @intFromFloat(x*distance);
+					const yOffset: i32 = @intFromFloat(y*distance);
+					const zOffset: i32 = @intFromFloat(z*distance);
+
+					const p = pos + Vec3i{xOffset, yOffset, zOffset};
+
+					var oldBlock = main.server.world.?.getBlock(p[0], p[1], p[2]) orelse continue;
+					if(oldBlock.typ == 0) continue;
+
+					if(damage < oldBlock.blockResistance()) break;
+					damage -= oldBlock.blockHealth();
```

Comment:
This would be more in line with how block breaking works.
```suggestion
					damage -= oldBlock.blockHealth() + oldBlock.blockResistance();
```